src/test/scala/is/hail/expr/ir/FunctionSuite.scala

```diff
@@ -0,0 +1,107 @@
+package is.hail.expr.ir
+
+import java.io.PrintWriter
+
+import is.hail.annotations._
+import is.hail.asm4s._
+import is.hail.expr.ir.functions.{IRFunctionRegistry, RegistryFunctions, UtilFunctions}
+import is.hail.expr.types._
+import org.testng.annotations.Test
+import is.hail.expr.{EvalContext, FunType, Parser}
+
+import scala.reflect.ClassTag
+
+object ScalaTestObject {
+  def testFunction(): Int = 1
+}
+
+object ScalaTestCompanion {
+  def testFunction(): Int = 1
+}
+
+class ScalaTestCompanion {
+  def testFunction(): Int = 1
```

Comment:
Might be good to make these return different values to make sure you're calling the right one.